SIGFPE when loading SpatiaLite on OS X

When loading SpatiaLite on OS X, the following error occurs. This doesn't happen on Linux. Using SpatiaLite installed from Homebrew

```
fatal error: unexpected signal during runtime execution
[signal SIGFPE: floating-point exception code=0x7 addr=0x85c0426 pc=0x85c0426]

runtime stack:
runtime.throw(0x4541889, 0x2a)
	/usr/local/Cellar/go/1.14.4/libexec/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/usr/local/Cellar/go/1.14.4/libexec/src/runtime/signal_unix.go:679 +0x46a

goroutine 7 [syscall]:
runtime.cgocall(0x4354b60, 0xc00014ec10, 0xc00014ec20)
	/usr/local/Cellar/go/1.14.4/libexec/src/runtime/cgocall.go:133 +0x5b fp=0xc00014ebe0 sp=0xc00014eba8 pc=0x400536b
github.com/mattn/go-sqlite3._Cfunc_sqlite3_load_extension(0x8004100, 0x8005b80, 0x8005770, 0x0, 0x0)
	_cgo_gotypes.go:938 +0x4d fp=0xc00014ec10 sp=0xc00014ebe0 pc=0x431a95d
github.com/mattn/go-sqlite3.(*SQLiteConn).LoadExtension.func5(0xc000062420, 0x8005b80, 0x8005770, 0xffffffffffffffff)
	/Users/lstoll/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_load_extension.go:59 +0x73 fp=0xc00014ec50 sp=0xc00014ec10 pc=0x4330ee3
github.com/mattn/go-sqlite3.(*SQLiteConn).LoadExtension(0xc000062420, 0x453523a, 0xe, 0x453a2a9, 0x1a, 0x0, 0x0)
	/Users/lstoll/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_load_extension.go:59 +0x18a fp=0xc00014ecd8 sp=0xc00014ec50 pc=0x432b65a
github.com/lstoll/wherewasi.registerSpatiaLite.func1(0xc000062420, 0x1c, 0x0)
	/Users/lstoll/src/wherewasi/main.go:214 +0xd8 fp=0xc00014ed80 sp=0xc00014ecd8 pc=0x434f818
github.com/mattn/go-sqlite3.(*SQLiteDriver).Open(0xc00000e300, 0xc00001a0f0, 0x4a, 0x0, 0x0, 0x0, 0x0)
	/Users/lstoll/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3.go:1660 +0x1873 fp=0xc00014f370 sp=0xc00014ed80 pc=0x43248e3
database/sql.dsnConnector.Connect(...)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:688
database/sql.(*dsnConnector).Connect(0xc00000e500, 0x45c32e0, 0xc0000140d0, 0x0, 0x0, 0x0, 0xc00004c430)
	<autogenerated>:1 +0x53 fp=0xc00014f3b8 sp=0xc00014f370 pc=0x415ba53
database/sql.(*DB).conn(0xc00015c0c0, 0x45c32e0, 0xc0000140d0, 0x1, 0x485f720, 0x4970108, 0x0)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1228 +0x201 fp=0xc00014f570 sp=0xc00014f3b8 pc=0x414ee61
database/sql.(*DB).exec(0xc00015c0c0, 0x45c32e0, 0xc0000140d0, 0x45481b8, 0x67, 0x0, 0x0, 0x0, 0xc00000e501, 0x20, ...)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1495 +0x66 fp=0xc00014f600 sp=0xc00014f570 pc=0x41510c6
database/sql.(*DB).ExecContext(0xc00015c0c0, 0x45c32e0, 0xc0000140d0, 0x45481b8, 0x67, 0x0, 0x0, 0x0, 0x20, 0x44e7860, ...)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1477 +0xde fp=0xc00014f6a0 sp=0xc00014f600 pc=0x4150dae
github.com/lstoll/wherewasi.(*Storage).migrate(0xc00000e520, 0x45c32e0, 0xc0000140d0, 0x4a, 0xc00015c0c0)
	/Users/lstoll/src/wherewasi/sql.go:310 +0x6f fp=0xc00014f710 sp=0xc00014f6a0 pc=0x434c0df
github.com/lstoll/wherewasi.newStorage(0x45c32e0, 0xc0000140d0, 0x45c34e0, 0xc00006c190, 0xc00001a0f0, 0x4a, 0x0, 0x0, 0x0)
	/Users/lstoll/src/wherewasi/sql.go:302 +0x161 fp=0xc00014f778 sp=0xc00014f710 pc=0x434bfd1
github.com/lstoll/wherewasi.setupDB(0xc00011ea20, 0x45c32e0, 0xc0000140d0, 0x0)
	/Users/lstoll/src/wherewasi/sql_test.go:58 +0x312 fp=0xc00014f830 sp=0xc00014f778 pc=0x434ec52
github.com/lstoll/wherewasi.TestUpsert4sqCheckin(0xc00011ea20)
	/Users/lstoll/src/wherewasi/sql_checkin_test.go:6 +0x43 fp=0xc00014ff80 sp=0xc00014f830 pc=0x434d893
testing.tRunner(0xc00011ea20, 0x454fc38)
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:991 +0xdc fp=0xc00014ffd0 sp=0xc00014ff80 pc=0x40f95ec
runtime.goexit()
	/usr/local/Cellar/go/1.14.4/libexec/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00014ffd8 sp=0xc00014ffd0 pc=0x4068221
created by testing.(*T).Run
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:1042 +0x357

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00011ea20, 0x4537613, 0x14, 0x454fc38, 0x4089901)
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:1043 +0x37e
testing.runTests.func1(0xc00011e6c0)
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:1284 +0x78
testing.tRunner(0xc00011e6c0, 0xc00004be10)
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:991 +0xdc
testing.runTests(0xc00000e320, 0x4856500, 0x5, 0x5, 0x0)
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:1282 +0x2a7
testing.(*M).Run(0xc000142000, 0x0)
	/usr/local/Cellar/go/1.14.4/libexec/src/testing/testing.go:1199 +0x15f
main.main()
	_testmain.go:52 +0x135

goroutine 8 [select]:
database/sql.(*DB).connectionOpener(0xc00015c000, 0x45c32a0, 0xc000012980)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1052 +0xe8
created by database/sql.OpenDB
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:722 +0x15d

goroutine 9 [select]:
database/sql.(*DB).connectionResetter(0xc00015c000, 0x45c32a0, 0xc000012980)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1065 +0xfb
created by database/sql.OpenDB
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:723 +0x193

goroutine 10 [select]:
database/sql.(*DB).connectionOpener(0xc00015c0c0, 0x45c32a0, 0xc0000129c0)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1052 +0xe8
created by database/sql.OpenDB
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:722 +0x15d

goroutine 11 [select]:
database/sql.(*DB).connectionResetter(0xc00015c0c0, 0x45c32a0, 0xc0000129c0)
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:1065 +0xfb
created by database/sql.OpenDB
	/usr/local/Cellar/go/1.14.4/libexec/src/database/sql/sql.go:723 +0x193
FAIL	github.com/lstoll/wherewasi	0.349s
FAIL
make: *** [test] Error 1
```

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -4,7 +4,7 @@ jobs:
 
   build:
     name: Build
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     steps:
 
     - name: Set up Go 1.14

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -14,7 +14,7 @@ jobs:
       id: go
 
     - name: Install dependencies
-      run: sudo apt-get update && sudo apt-get install -y libsqlite3-dev
+      run: sudo apt-get update && sudo apt-get install -y libsqlite3-dev libspatialite-dev
 
     - name: Check out code into the Go module directory
       uses: actions/checkout@v1

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -13,6 +13,9 @@ jobs:
         go-version: 1.14
       id: go
 
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y libsqlite3-dev
+
     - name: Check out code into the Go module directory
       uses: actions/checkout@v1
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -3,7 +3,7 @@ FROM golang:1.14-buster AS build
 WORKDIR /src
 
 RUN apt-get update && \
-    apt-get install -y libspatialite-dev
+    apt-get install -y libsqlite3-dev libspatialite-dev
 
 COPY go.mod go.sum ./
 RUN go mod download
@@ -14,7 +14,7 @@ RUN go install ./...
 FROM debian:buster
 
 RUN apt-get update && \
-    apt-get install -y sqlite3 libspatialite7 ca-certificates
+    apt-get install -y sqlite3 libsqlite3-0 libspatialite7 ca-certificates
 
 COPY --from=build /go/bin/wherewasi /usr/bin/
 

--- a/Makefile
+++ b/Makefile
@@ -2,13 +2,15 @@
 
 GOPATH=$(shell go env GOPATH)
 
+go_build_flags="-tags=libsqlite3"
+
 all: $(GOPATH)/bin/wherewasi test lint
 
 $(GOPATH)/bin/wherewasi: *
-	go install .
+	go install $(go_build_flags) .
 
 test:
-	go test -v .
+	go test $(go_build_flags) -v .
 
 lint: bin/golangci-lint-1.23.8
 	./bin/golangci-lint-1.23.8 run ./...

--- a/main.go
+++ b/main.go
@@ -198,13 +198,9 @@ func registerSpatiaLite() {
 	exts := map[string]string{}
 
 	if runtime.GOOS == "linux" {
-		// exts["libspatialite.so.7"] = "spatialite_init_ex"
+		exts["libspatialite.so.7"] = "spatialite_init_ex"
 	} else if runtime.GOOS == "darwin" {
-		// Disabling for now, throws
-		// [signal SIGFPE: floating-point exception code=0x7 addr=0x6d65426 pc=0x6d65426]
-
-		// exts["mod_spatialite"] = "sqlite3_modspatialite_init"
-		_ = struct{}{}
+		exts["mod_spatialite"] = "sqlite3_modspatialite_init"
 	}
 
 	sql.Register("spatialite", &sqlite3.SQLiteDriver{

